.github/workflow/deploy.yml

```diff
@@ -0,0 +1,36 @@
+name: deploy-book
+
+# Only run this when the master branch changes
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
```

Comment:
```suggestion
# Only run this when the master branch changes
on:
  push:
    branches:
    - master
```

If you want your comment to be true, you need to remove the Pull Request trigger otherwise the job will also run when PRs to the master branch are opened. Unless that is what you want?

Reply:
@sgibson91, thanks for the revision! I would say it's ok to run the job only for push. @Delphine-L and I had a recent discussion today of the listed branches, when is useful to declare more than a single branch?

Finally, for the issues and PR templates, are they ok for you? They are an adaptation from this repo, 
https://github.com/yadplayz/musicbot2345/tree/1aff2ef0aae93f13a3a014b78fdea101422feba3/.github/ISSUE_TEMPLATE
I've also found more sophisticated ones, but using `yml` instead of `md`.  Which format would you recommend?
https://github.com/theos/theos/blob/master/.github/ISSUE_TEMPLATE

Reply:
> when is useful to declare more than a single branch?

I would say in very software development cases where a strict gitflow is being followed. There may be 'special' branches beyond `main`/`master`, such as `staging` and `prod` or `features/**`, where you will always want something like a test suite to run when those branches are pushed to. It's definitely overkill for this project :)

> Finally, for the issues and PR templates, are they ok for you?

Yes, they look great!

> I've also found more sophisticated ones, but using yml instead of md. Which format would you recommend?

This is personal preference, especially since the YAML formatted templates are still a beta feature of GitHub. The YAML ones render more like a form, you can see an example here: https://github.com/2i2c-org/infrastructure/issues/new?assignees=&labels=type%3A+hub&template=2_new-hub.yml&title=%5BNew+Hub%5D+%7B%7B+HUB+NAME+%7D%7D ([raw format](https://raw.githubusercontent.com/2i2c-org/infrastructure/master/.github/ISSUE_TEMPLATE/2_new-hub.yml))
These can feel a bit easier for folks who are new to GitHub to fill in, but that's dependent on your community.

Docs on how to set up issue forms if you choose to go down that route are here: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

Reply:
wowwww thanks for going through each of my questions. I have now plenty of useful resources to continue the integration of templates and CI within the EnvAI book and future projects.